File: core/java/src/main/java/com/vxlisp/vx/translation/Es.java
package com.vxlisp.vx.translation;

import java.util.LinkedHashMap;
import java.util.Map;
import com.vxlisp.vx.*;

public final class Es {

  /**
   * @function translation_es
   * Returns translation data
   * @return {translation}
   * (func translation-es)
   */
  public interface Func_translation_es extends Core.Type_func, Core.Type_replfunc {
    public Core.Type_translation vx_translation_es();
  }

  public static class Class_translation_es extends Core.Class_base implements Func_translation_es {

    @Override
    public Es.Func_translation_es vx_new(final Object... vals) {
      Class_translation_es output = new Class_translation_es();
      return output;
    }

    @Override
    public Es.Func_translation_es vx_copy(final Object... vals) {
      Class_translation_es output = new Class_translation_es();
      return output;
    }

    @Override
    public Core.Type_typedef vx_typedef() {
      return Core.t_func.vx_typedef();
    }

    @Override
    public Core.Type_funcdef vx_funcdef() {
      return Core.funcdef_new(
        "vx/translation/es", // pkgname
        "translation-es", // name
        0, // idx
        false, // async
        Core.typedef_new(
          "vx/core", // pkgname
          "translation", // name
          ":struct", // extends
          Core.e_typelist, // traits
          Core.e_typelist, // allowtypes
          Core.e_typelist, // disallowtypes
          Core.e_funclist, // allowfuncs
          Core.e_funclist, // disallowfuncs
          Core.e_anylist, // allowvalues
          Core.e_anylist, // disallowvalues
          Core.e_argmap // properties
        ) // typedef
      );
    }

    @Override
    public Core.Type_any vx_empty() {
      return e_translation_es;
    }

    @Override
    public Core.Type_any vx_type() {
      return t_translation_es;
    }

    public Core.Type_any vx_repl(Core.Type_anylist arglist) {
      Core.Type_any output = Core.e_any;
      output = Es.f_translation_es();
      return output;
    }

    @Override
    public Core.Type_translation vx_translation_es() {
      return Es.f_translation_es();
    }

  }

  public static final Func_translation_es e_translation_es = new Es.Class_translation_es();
  public static final Func_translation_es t_translation_es = new Es.Class_translation_es();

  public static Core.Type_translation f_translation_es() {
    Core.Type_translation output = Core.e_translation;
    output = Core.f_new(
      Core.t_translation,
      Core.vx_new(Core.t_anylist,
        Core.vx_new_string(":name"),
        Core.vx_new_string("es"),
        Core.vx_new_string(":wordmap"),
        Es.f_words()
      )
    );
    return output;
  }

  /**
   * @function words
   * Word Translation
   * @return {stringmap}
   * (func words)
   */
  public interface Func_words extends Core.Type_func, Core.Type_replfunc {
    public Core.Type_stringmap vx_words();
  }

  public static class Class_words extends Core.Class_base implements Func_words {

    @Override
    public Es.Func_words vx_new(final Object... vals) {
      Class_words output = new Class_words();
      return output;
    }

    @Override
    public Es.Func_words vx_copy(final Object... vals) {
      Class_words output = new Class_words();
      return output;
    }

    @Override
    public Core.Type_typedef vx_typedef() {
      return Core.t_func.vx_typedef();
    }

    @Override
    public Core.Type_funcdef vx_funcdef() {
      return Core.funcdef_new(
        "vx/translation/es", // pkgname
        "words", // name
        0, // idx
        false, // async
        Core.typedef_new(
          "vx/core", // pkgname
          "stringmap", // name
          ":map", // extends
          Core.e_typelist, // traits
          Core.vx_new(Core.t_typelist, Core.t_string), // allowtypes
          Core.e_typelist, // disallowtypes
          Core.e_funclist, // allowfuncs
          Core.e_funclist, // disallowfuncs
          Core.e_anylist, // allowvalues
          Core.e_anylist, // disallowvalues
          Core.e_argmap // properties
        ) // typedef
      );
    }

    @Override
    public Core.Type_any vx_empty() {
      return e_words;
    }

    @Override
    public Core.Type_any vx_type() {
      return t_words;
    }

    public Core.Type_any vx_repl(Core.Type_anylist arglist) {
      Core.Type_any output = Core.e_any;
      output = Es.f_words();
      return output;
    }

    @Override
    public Core.Type_stringmap vx_words() {
      return Es.f_words();
    }

  }

  public static final Func_words e_words = new Es.Class_words();
  public static final Func_words t_words = new Es.Class_words();

  public static Core.Type_stringmap f_words() {
    Core.Type_stringmap output = Core.e_stringmap;
    output = Core.f_new(
      Core.t_stringmap,
      Core.vx_new(Core.t_anylist,
        Core.vx_new_string("Hello"),
        Core.vx_new_string("Hola"),
        Core.vx_new_string("World"),
        Core.vx_new_string("Mundo")
      )
    );
    return output;
  }


  static {
    Map<String, Core.Type_any> maptype = new LinkedHashMap<String, Core.Type_any>();
    Map<String, Core.Type_any> mapconst = new LinkedHashMap<String, Core.Type_any>();
    Map<String, Core.Type_func> mapfunc = new LinkedHashMap<String, Core.Type_func>();
    mapfunc.put("translation-es", Es.t_translation_es);
    mapfunc.put("words", Es.t_words);
    Core.vx_global_package_set("vx/translation/es", maptype, mapconst, mapfunc);
  }

}

File: core/kotlin/app/src/main/kotlin/com/vxlisp/vx/translation/Es.kt
package com.vxlisp.vx.translation

import com.vxlisp.vx.*

object vx_translation_es {


  interface Func_translation_es : vx_core.Type_func, vx_core.Type_replfunc {
    fun vx_translation_es() : vx_core.Type_translation
  }

  class Class_translation_es : vx_core.Class_base, Func_translation_es {
    constructor() {}

    override fun vx_new(vararg vals : Any) : vx_core.Type_any {
      val output : vx_translation_es.Class_translation_es = vx_translation_es.Class_translation_es()
      return output
    }

    override fun vx_copy(vararg vals : Any) : vx_core.Type_any {
      val output : vx_translation_es.Class_translation_es = vx_translation_es.Class_translation_es()
      return output
    }

    override fun vx_typedef() : vx_core.Type_typedef {
      var output : vx_core.Type_typedef = vx_core.t_func.vx_typedef()
      return output
    }

    override fun vx_funcdef() : vx_core.Type_funcdef {
      var output : vx_core.Type_funcdef = vx_core.funcdef_new(
        "vx/translation/es", // pkgname
        "translation-es", // name
        0, // idx
        false, // async
        vx_core.typedef_new(
          "vx/core", // pkgname
          "translation", // name
          ":struct", // extends
          vx_core.e_typelist, // traits
          vx_core.e_typelist, // allowtypes
          vx_core.e_typelist, // disallowtypes
          vx_core.e_funclist, // allowfuncs
          vx_core.e_funclist, // disallowfuncs
          vx_core.e_anylist, // allowvalues
          vx_core.e_anylist, // disallowvalues
          vx_core.e_argmap // properties
        ) // typedef
      )
      return output
    }

    override fun vx_empty() : vx_core.Type_any {
      var output : vx_core.Type_any = vx_translation_es.e_translation_es
      return output
    }

    override fun vx_type() : vx_core.Type_any {
      var output : vx_core.Type_any = vx_translation_es.t_translation_es
      return output
    }

    override fun vx_repl(arglist : vx_core.Type_anylist) : vx_core.Type_any {
      var output : vx_core.Type_any = vx_core.e_any
      output = vx_translation_es.f_translation_es()
      return output
    }

    override fun vx_translation_es() : vx_core.Type_translation {
      var output : vx_core.Type_translation = vx_translation_es.f_translation_es()
      return output
    }

  }

  val e_translation_es : vx_translation_es.Func_translation_es = vx_translation_es.Class_translation_es()
  val t_translation_es : vx_translation_es.Func_translation_es = vx_translation_es.Class_translation_es()

  fun f_translation_es() : vx_core.Type_translation {
    var output : vx_core.Type_translation = vx_core.e_translation
    output = vx_core.f_new(
      vx_core.t_translation,
      vx_core.vx_new(
        vx_core.t_anylist,
        vx_core.vx_new_string(":name"),
        vx_core.vx_new_string("es"),
        vx_core.vx_new_string(":wordmap"),
        vx_translation_es.f_words()
      )
    )
    return output
  }


  interface Func_words : vx_core.Type_func, vx_core.Type_replfunc {
    fun vx_words() : vx_core.Type_stringmap
  }

  class Class_words : vx_core.Class_base, Func_words {
    constructor() {}

    override fun vx_new(vararg vals : Any) : vx_core.Type_any {
      val output : vx_translation_es.Class_words = vx_translation_es.Class_words()
      return output
    }

    override fun vx_copy(vararg vals : Any) : vx_core.Type_any {
      val output : vx_translation_es.Class_words = vx_translation_es.Class_words()
      return output
    }

    override fun vx_typedef() : vx_core.Type_typedef {
      var output : vx_core.Type_typedef = vx_core.t_func.vx_typedef()
      return output
    }

    override fun vx_funcdef() : vx_core.Type_funcdef {
      var output : vx_core.Type_funcdef = vx_core.funcdef_new(
        "vx/translation/es", // pkgname
        "words", // name
        0, // idx
        false, // async
        vx_core.typedef_new(
          "vx/core", // pkgname
          "stringmap", // name
          ":map", // extends
          vx_core.e_typelist, // traits
          vx_core.vx_new(vx_core.t_typelist, vx_core.t_string), // allowtypes
          vx_core.e_typelist, // disallowtypes
          vx_core.e_funclist, // allowfuncs
          vx_core.e_funclist, // disallowfuncs
          vx_core.e_anylist, // allowvalues
          vx_core.e_anylist, // disallowvalues
          vx_core.e_argmap // properties
        ) // typedef
      )
      return output
    }

    override fun vx_empty() : vx_core.Type_any {
      var output : vx_core.Type_any = vx_translation_es.e_words
      return output
    }

    override fun vx_type() : vx_core.Type_any {
      var output : vx_core.Type_any = vx_translation_es.t_words
      return output
    }

    override fun vx_repl(arglist : vx_core.Type_anylist) : vx_core.Type_any {
      var output : vx_core.Type_any = vx_core.e_any
      output = vx_translation_es.f_words()
      return output
    }

    override fun vx_words() : vx_core.Type_stringmap {
      var output : vx_core.Type_stringmap = vx_translation_es.f_words()
      return output
    }

  }

  val e_words : vx_translation_es.Func_words = vx_translation_es.Class_words()
  val t_words : vx_translation_es.Func_words = vx_translation_es.Class_words()

  fun f_words() : vx_core.Type_stringmap {
    var output : vx_core.Type_stringmap = vx_core.e_stringmap
    output = vx_core.f_new(
      vx_core.t_stringmap,
      vx_core.vx_new(
        vx_core.t_anylist,
        vx_core.vx_new_string("Hello"),
        vx_core.vx_new_string("Hola"),
        vx_core.vx_new_string("World"),
        vx_core.vx_new_string("Mundo")
      )
    )
    return output
  }


  init {
    var maptype : MutableMap<String, vx_core.Type_any> = LinkedHashMap<String, vx_core.Type_any>()
    var mapconst : MutableMap<String, vx_core.Type_any> = LinkedHashMap<String, vx_core.Type_any>()
    var mapfunc : MutableMap<String, vx_core.Type_func> = LinkedHashMap<String, vx_core.Type_func>()
    mapfunc.put("translation-es", vx_translation_es.t_translation_es)
    mapfunc.put("words", vx_translation_es.t_words)
    vx_core.vx_global_package_set("vx/translation/es", maptype, mapconst, mapfunc)
  }

}
